### PR TITLE
Support application/wasm in Docker image

### LINKED ISSUE #11857: Docker image doesn't support serving application/wasm
Which breaks compiling wasm, which means some of our workers are slower apparently.

Related: https://github.com/matrix-org/matrix-hosted/issues/1096

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #11857

### CODE REVIEW

**Comment by t3chguy** [comment]
The branch changes seem unrelated, whats the justification for them?

**Comment by Half-Shot** [comment]
>The branch changes seem unrelated, whats the justification for them?

Me being an idiot and committing them :)

File: Dockerfile [hunk 1]
```diff
@@ -30,5 +30,11 @@ FROM nginx:alpine
 
 COPY --from=builder /src/webapp /app
 
+# Insert wasm type into Nginx mime.types file so they load correctly.
+RUN sed '3i\ \ \ \ application/wasm wasm\;' /etc/nginx/mime.types > /tmp/mime.types
+RUN cp /tmp/mime.types /etc/nginx/mime.types
+
+RUN cat /etc/nginx/mime.types
```
Comment (turt2live) [inline Dockerfile]: we can remove this

File: Dockerfile [hunk 2]
```diff
@@ -30,5 +30,11 @@ FROM nginx:alpine
 
 COPY --from=builder /src/webapp /app
 
+# Insert wasm type into Nginx mime.types file so they load correctly.
+RUN sed '3i\ \ \ \ application/wasm wasm\;' /etc/nginx/mime.types > /tmp/mime.types
```
Comment (turt2live) [inline Dockerfile]: any reason we can't pipe directly to the nginx config?

Comment (Half-Shot) [reply Dockerfile]: I found out today that `sed` operates line by line, so if you `sed $thing a > a` you will typically end up with a blank file.

Comment (t3chguy) [reply Dockerfile]: you could `echo ... >> ...` (>> being append) though, no?

Comment (Half-Shot) [reply Dockerfile]: Apparently you can even do `-i`. @babolivier is a wizard and he knows better than all of StackOverflow.

Comment (babolivier) [reply Dockerfile]: brb printing that on a tee shirt.

**Pull request merged**